packages/components/src/geolocation/GeolocationProvider.tsx

```diff
@@ -0,0 +1,112 @@
+import { useTranslation } from 'next-i18next';
+import { createContext, useContext, useState, useCallback } from 'react';
+import type { ReactNode, Context } from 'react';
+import type { Coordinates } from '../types';
+import sendNotification from '../utils/sendToast';
+import geolocationService from './GeolocationService';
+
+export type GeolocationContextType = {
+  coordinates: Coordinates | null;
+  loading: boolean;
+  called: boolean;
+  error?: GeolocationPositionError | null;
+  resolve: () => Promise<Coordinates | null>;
+};
+
+const GeolocationContext: Context<GeolocationContextType> =
+  createContext<GeolocationContextType>({
+    coordinates: null,
+    loading: false,
+    called: false,
+    error: null,
+    resolve: () => Promise.resolve(null),
+  });
+
+type Props = {
+  children: ReactNode;
+};
+
+export const isGeolocationPositionError = (
+  error: unknown
+): error is GeolocationPositionError =>
+  Boolean(
+    error &&
+      typeof error === 'object' &&
+      'code' in error &&
+      'message' in error &&
+      'PERMISSION_DENIED' in error &&
+      'POSITION_UNAVAILABLE' in error &&
+      'TIMEOUT' in error
+  );
+
+export default function GeolocationProvider({ children }: Props) {
+  const { t } = useTranslation('geolocationProvider');
+  const [location, setLocation] = useState<Coordinates | null>(null);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [called, setCalled] = useState<boolean>(false);
+  const [error, setError] = useState<GeolocationPositionError | null>(null);
+
+  const resolve = useCallback(async () => {
+    setLoading(true);
+
+    if (!('geolocation' in navigator)) {
+      setLocation(null);
+      setLoading(false);
+      setCalled(true);
+
+      return null;
+    }
+
+    let geolocation = null;
+
+    sendNotification({
+      title: t('a11ymessage.title'),
+      message: t('a11ymessage.description'),
+      type: 'info',
+      options: { invisible: true },
+    });
+
+    try {
```

Comment:
Should this be in useEffect hook so that it wont be tried to run on server side render?

Reply:
useEffect was in other place, so this was not issue.